keras/src/layers/layer_test.py

```diff
@@ -1544,3 +1544,111 @@ def call(self, inputs):
         layer = MyDenseLayer(10)
         output = layer(inputs)
         self.assertAllEqual(output.shape, (10, 10))
+
+    def test_call_context_args_with_custom_layers(self):
+        class Inner(layers.Layer):
+            call_context_args = ("foo_mode",)
+
+            def call(self, x, foo_mode=None):
+                return x + (1 if foo_mode else 0)
+
+        class Outer(layers.Layer):
+            call_context_args = ("foo_mode",)
+
+            def __init__(self):
+                super().__init__()
+                self.inner = Inner()
+
+            def call(self, x):
+                # Outer doesn’t even need to re‑inject explicitly:
+                # our base class will propagate foo_mode automatically
+                return self.inner(x)
+
+        layer = Outer()
+        self.assertEqual(int(layer(np.array(0), foo_mode=True)), 1)
+        self.assertEqual(int(layer(np.array(0))), 0)
+
+    def test_context_args_with_triple_nesting_and_priority(self):
+        class Inner(layers.Layer):
+            call_context_args = ("foo_mode",)
+
+            def call(self, x, foo_mode=None):
+                return x + (1 if foo_mode else 0)
+
+        class Middle(layers.Layer):
+            call_context_args = ("foo_mode",)
```

Comment:
Can you remove this line and `foo_mode` in `call` below, does it still work?

I'd like to have a test with an intermediate layer that doesn't handle `foo_mode` at all to make sure that works.

Reply:
Done, verified it still works.

However, the outermost layer still needs to declare `call_context_args = ("foo_mode",)`. Or should even that not be required?

Reply:
It's required, right? Otherwise it wouldn't be put in the CallContext.
I'm fine with that.